src/models/docdbUtils.js

```diff
@@ -1,57 +1,80 @@
+/*
+** DocumentDB Utilities
+**  - Useful, reusable code for our application
+*/
+"use strict";
+
+
 var DocumentDBClient = require('documentdb').DocumentClient;
 
 var DocDBUtils = {
-  getOrCreateDatabase: function(client, databaseId, callback) {
-    var querySpec = {
-      query: 'SELECT * FROM root r WHERE r.id=@id',
-      parameters: [{
-        name: '@id',
-        value: databaseId
-      }]
-    };
-
-    client.queryDatabases(querySpec).toArray(function(err, results) {
-      if (err) {
-        callback(err);
-      }
-
-      if (!err && results.length === 0) {
-        client.createDatabase({
-          id: databaseId
-        }, function(err, created) {
-          callback(null, created);
+    // getOrCreateDatabase
+    //  Connect to existing database or create it (by databaseId)
+    getOrCreateDatabase: function (client, databaseId, callback) {
+        var querySpec = {
+            query: 'SELECT * FROM root r WHERE r.id= @id',
+            parameters: [{
+                name: '@id',
+                value: databaseId
+            }]
+        };
+
+        client.queryDatabases(querySpec).toArray(function (err, results) {
+            if (err) {
+                callback(err);
+
+            } else {
+                if (results.length === 0) {
+                    var databaseSpec = {
+                        id: databaseId
+                    };
+
+                    client.createDatabase(databaseSpec, function (err, created) {
+                        callback(null, created);
+                    });
+
+                } else {
+                    callback(null, results[0]);
+                }
+            }
         });
-      } else {
-        callback(null, results[0]);
-      }
-    });
-  },
-
-  getOrCreateCollection: function(client, databaseLink, collectionId, callback) {
-    var querySpec = {
-      query: 'SELECT * FROM root r WHERE r.id=@id',
-      parameters: [{
-        name: '@id',
-        value: collectionId
-      }]
-    };
-
-    client.queryCollections(databaseLink, querySpec).toArray(function(err, results) {
-      if (err) {
-        callback(err);
-      }
-
-      if (!err && results.length === 0) {
-        client.createCollection(databaseLink, {
-          id: collectionId
-        }, function(err, created) {
-          callback(null, created);
+    },
+
+    // getOrCreateCollection
+    //  Connect to existing collection or create it (by collectionId)
+    getOrCreateCollection: function (client, databaseLink, collectionId, callback) {
```

Comment:
Not your change, but we should update the sample to use databaseId instead of databaseLink (old API)